### PR TITLE
remove FairTask usage fro reconstruction

### PR DESCRIPTION
**Pull request by AllaMaevskaya**
remove FairTask usage fro reconstruction

### CODE REVIEW

File: Detectors/FIT/simulation/include/FITSimulation/#DigitizerTask.h# [hunk 1]
```diff
@@ -0,0 +1,80 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
```
Comment (sawenzel) [inline Detectors/FIT/simulation/include/FITSimulation/#DigitizerTask.h#]: this file should not be committed.

File: Detectors/FIT/simulation/include/FITSimulation/Digitizer.h [hunk 2]
```diff
@@ -35,28 +35,33 @@ class Digitizer
   void setEventTime(double value) { mEventTime = value; }
   void setEventID(Int_t id) { mEventID = id; }
   void setSrcID(Int_t id) { mSrcID = id; }
-  Int_t getCurrentTimeFrame() const { return mTimeFrameCurrent; }
-  void setCurrentTimeFrame(Double_t value) { mTimeFrameCurrent = value; }
+  void setBC(Int_t bc) { mBC = bc; }
+  void setOrbit(Int_t orbit) { mOrbit = orbit; }
+
+  void setTriggers(Digit* digit);
+  void smearCFDtime(Digit* digit);
```
Comment (sawenzel) [inline Detectors/FIT/simulation/include/FITSimulation/Digitizer.h]: do you modify the digit in this function? If not, consider passing digit as `const& `. Applies also to `setTriggers.`

File: Detectors/FIT/simulation/include/FITSimulation/#DigitizerTask.h# (hunk 1 above)
Comment (AllaMaevskaya) [reply Detectors/FIT/simulation/include/FITSimulation/#DigitizerTask.h#]: sure. Sorry

**Comment by AllaMaevskaya** [comment]
yes, i modify digits there
________________________________________

**Comment by AllaMaevskaya** [comment]
Hello

- in process method for each channel average time from all hits and sources was calculated;
- in smearCFDtime(Digit* digits) time signals are smeared by 50ps. In each channel average time from all sources 
- in setTriggers trigger signals (based on time and amplitude information from all hits sources) are added to digits;
So digits are modified in both setTriggers and smearCFDtime

Best regards
Alla

File: Detectors/FIT/simulation/src/Digitizer.cxx [hunk 3]
```diff
@@ -82,127 +73,118 @@ void Digitizer::process(const std::vector<HitType>* hits, Digit* digit)
...
-    //if ampl less than cfd trh adc and cfd has no signal
-    if (ch_signal_MIP[ch_iter] < CFD_trsh_mip) {
-      ch_signal_MIP[ch_iter] = 0.;
-      ch_signal_time[ch_iter] = 0.;
+//------------------------------------------------------------------------
+void Digitizer::smearCFDtime(Digit* digit)
+{
+  //smeared CFD time for 50ps
+  std::vector<ChannelData> mChDgDataArr;
+  for (const auto& d : digit->getChDgData()) {
```
Comment (shahor02) [inline Detectors/FIT/simulation/src/Digitizer.cxx]: I don't understand this code: you smear only the stack mChDgDataArr vector, w/o any effect on the digit or the digitizer state

File: Detectors/FIT/reconstruction/src/RecPoints.cxx [hunk 4]
```diff
@@ -29,24 +30,22 @@ void RecPoints::FillFromDigits(const Digit& digit)
...
+  mEventTime = digit.getTime();
+  mBC = digit.getBC();
+  mOrbit = digit.getOrbit();
+
+  Float_t BCEventTime = 12.5;
+
   for (const auto& d : digit.getChDgData()) {
     Int_t mcp = d.ChId;
-    cfd[mcp] = d.CFDTime /*- BC_clk_center - BCEventTime*/;
+    cfd[mcp] = d.CFDTime - mEventTime - BCEventTime;
```
Comment (shahor02) [inline Detectors/FIT/reconstruction/src/RecPoints.cxx]: Does the mEventTime here correspond to the MC-truth time supplied by the interaction sampler during the digitization? The reconstruction should not use MC-truth times, should not you derive it from the orbit and BC?

Comment (AllaMaevskaya) [reply Detectors/FIT/reconstruction/src/RecPoints.cxx]: Dear Ruben and Sandro,
I fixed code according your suggestion.
May I ask to merge branch now? Next step will be to separate T0 and V0 within FIT directory. 
We need to start from recent dev branch with all these developments inside.

Best regards
Alla

**Comment by AllaMaevskaya** [comment]
Dear Sandro,
i read instructions, installed latest clang and checked my code again. It shows nothing criminal.[Clang/latest-o2] ~/aliO2/O2 $> git clang-format HEAD~1
clang-format did not modify any files
When i am looking in details of "continuous-integration/travis-ci/pr — The Travis CI build failed" i see that it checked something different than my latest commits.  At least in /Steer/DigitizerWorkflow/src/FITDigitizerSpec.cxx all lines are properly aligned, nothing like
         mDigitizer.process(&hits, &digit);
- 	auto data = digit.getChDgData();
+        auto data = digit.getChDgData();
         LOG(INFO) << "Have " << data.size() << " fired channels ";
Can you tell me what is wrong and how i can fix this?

Thank you 
Alla

**Comment by sawenzel** [comment]
This is because HEAD~1 roughly means compare to something 1 commit ago. Since your PR contains 19 commits, you need to use HEAD~19 or specify some other commit as a reference (like latest commit on dev).

**Comment by AllaMaevskaya** [comment]
Ooooooo! Now I see. 

Thank you

**Pull request merged**